desk/lib/logs.hoon

```diff
@@ -0,0 +1,57 @@
+/-  *logs
+|%
+::
+++  fail-event
+  |=  [=term =tang]
+  ^-  $>(%fail log-event)
+  [%fail term tang]
+::
+++  log-fail
+  |=  [=wire our=ship event=$>(%fail log-event)]
+  ^-  card:agent:gall
+  [%pass wire %agent [our %logs] %poke log-action+!>([%log event])]
+::
+++  enjs
+  =,  format
+  |%
+  ++  tang
+    |=  t=^tang
+    ^-  $>(%a json)
+    ?~  t  a+~
+    =/  tame=(list tape)
+      %+  turn  t
+     (cork (cury wash [0 80]) zing)
+    ::XX posthog does not display newlines properly
+    :: s+(crip (zing `(list tape)`(join "\0a" tame)))
+    a+(turn tame tape:enjs)
+  ::
+  ++  d-co-2  (d-co:co 2)
+  ++  id-event
+    |=  id=^id-event
+    ^-  $>(%s json)
+    =+  date=(yore id)
+    =+  frac=(head f.t.date)
+    ::  convert urbit fractosecs to millisecs
+    ::
+    =+  mili=(div (mul frac 1.000) 65.563)
+    :-  %s
+    %-  crip
+    "{((d-co:co 4) y.date)}-{(d-co-2 m.date)}-{(d-co-2 d.t.date)}".
+    "T{(d-co-2 h.t.date)}:{(d-co-2 m.t.date)}:{(d-co-2 s.t.date)}".
+    ".{((d-co:co 3) mili)}"
```

Comment:
As per [posthog docs](https://posthog.com/docs/data/timestamps#recognized-formats) they also support ingesting unix seconds and millisecond timestamps. So you can just use `+time:enjs:format` here. (: